cluster/aks.go

```diff
@@ -323,178 +492,141 @@ func (c *AKSCluster) GetStatus() (*pkgCluster.GetClusterStatusResponse, error) {
+			}
+		}
 	}
+	return nil
+}
 
-	client.With(c.log)
-
-	clusterSshSecret, err := c.getSshSecret(c)
+// UpdateCluster updates the cluster in AKS
+func (c *AKSCluster) UpdateCluster(request *pkgCluster.UpdateClusterRequest, userID uint) error {
+	cc, err := c.getCloudConnection()
 	if err != nil {
-		return emperror.Wrap(err, "failed to retrieve SSH secret")
-	}
-
-	sshKey := secret.NewSSHKeyPair(clusterSshSecret)
-
-	// send separate requests because Azure not supports multiple nodepool modification
-	// Azure not supports adding and deleting nodepools
-	var nodePoolAfterUpdate []*model.AKSNodePoolModel
-	var updatedCluster *azureType.ResponseWithValue
-	if requestNodes := request.AKS.NodePools; requestNodes != nil {
-		for name, np := range requestNodes {
-			log := c.log.WithField("nodePool", name)
-			if existNodePool := c.getExistingNodePoolByName(name); np != nil && existNodePool != nil {
-				log.Debug("Updating nodepool")
-
-				count := int32(np.Count)
-
-				// create request model for aks-client
-				ccr := azureCluster.CreateClusterRequest{
-					Name:              c.modelCluster.Name,
-					Location:          c.modelCluster.Location,
-					ResourceGroup:     c.modelCluster.AKS.ResourceGroup,
-					KubernetesVersion: c.modelCluster.AKS.KubernetesVersion,
-					SSHPubKey:         sshKey.PublicKeyData,
-					EnableRBAC:        c.RbacEnabled(),
-					Profiles: []containerservice.ManagedClusterAgentPoolProfile{
-						{
-							Name:   &name,
-							Count:  &count,
-							VMSize: containerservice.VMSizeTypes(existNodePool.NodeInstanceType),
-						},
-					},
-				}
-
-				nodePoolAfterUpdate = append(nodePoolAfterUpdate, &model.AKSNodePoolModel{
-					ID:               existNodePool.ID,
-					CreatedAt:        time.Now(),
-					CreatedBy:        userId,
-					ClusterID:        existNodePool.ClusterID,
-					Name:             name,
-					Autoscaling:      np.Autoscaling,
-					NodeMinCount:     np.MinCount,
-					NodeMaxCount:     np.MaxCount,
-					Count:            np.Count,
-					NodeInstanceType: existNodePool.NodeInstanceType,
-				})
-
-				updatedCluster, err = c.updateWithPolling(client, &ccr)
-				if err != nil {
-					return emperror.Wrap(err, "cluster update failed")
-				}
-			} else {
-				log.Warning("No such nodepool found")
-			}
-		}
+		return emperror.Wrap(err, "failed to get cloud connection")
 	}
+	client := cc.GetManagedClustersClient()
 
-	if updatedCluster != nil {
-		c.modelCluster.AKS.NodePools = nodePoolAfterUpdate
-		c.azureCluster = &updatedCluster.Value
+	cluster, err := c.getAzureCluster()
+	if err != nil {
+		return emperror.Wrap(err, "failed to retrieve AKS cluster")
 	}
 
-	return nil
-}
+	// AKS currently only supports a single node pool
+	// For more info, see: https://feedback.azure.com/forums/914020-azure-kubernetes-service-aks/suggestions/34917127-support-multiple-node-pool
+	if len(request.AKS.NodePools) > 1 {
+		return fmt.Errorf("Too many node pools: %d", len(request.AKS.NodePools))
```

Comment:
This is actually not correct. AKS allows creating clusters with multiple nodepools. It doesn't allow adding new nodepools to or removing existing nodepools from a running cluster. Only the size (min, max, node count) of the current nodepools can be adjusted on a running cluster.